File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: list[list[int]], x: int) -> int:
        if (n := len(grid) * len(grid[0])) == 1:
            return 0

        arr: list[int] = []
        mod: int = -1
        for row in grid:
            for num in row:
                arr.append(num)
                if mod == -1:
                    mod = num % x
                elif num % x != mod:
                    return -1
        
        arr = sorted(arr)
        median: int = arr[n // 2 - (n % 2 == 0)]
        
        return sum(abs(num - median) // x for row in grid for num in row)

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.c
int compare(const void *a, const void *b) 
{
    return (*(int*)a - *(int*)b);
}


int minOperations(int** grid, int gridSize, int* gridColSize, int x) 
{
    int n = gridSize * (*gridColSize), nums[n] = {};
    for (int i = 0; i < gridSize; ++i)
        for (int j = 0; j < *gridColSize; ++j)
        {
            if (grid[i][j] % x != grid[0][0] % x)
                return -1;
            nums[i * (*gridColSize) + j] = grid[i][j];
        }
    
    qsort(nums, n, sizeof(int), compare); 
    int median = nums[n / 2 - (n % 2 == 0)];

    int operations = 0;
    for (int i = 0; i < n; ++i)
        operations += abs(nums[i] - median) / x;
    
    return operations;
}